### PR TITLE
overlord: use state to discover all installed snaps

### PR DESCRIPTION
**Pull request by zyga**

### CODE REVIEW

File: overlord/snapstate/snapstate.go [hunk 1]
```diff
@@ -285,3 +285,27 @@ func Set(s *state.State, name string, snapst *SnapState) {
...
+	var infos []*snap.Info
+	if err := s.Get("snaps", &stateMap); err != nil && err != state.ErrNoState {
+		return nil, err
+	}
+	for snapName, snapState := range stateMap {
+		if !snapState.Active {
+			continue
+		}
+		if len(snapState.Sequence) == 0 {
+			continue // XXX: is this possible?
```
Comment (niemeyer) [inline overlord/snapstate/snapstate.go]: Won't hurt to check.

Comment (niemeyer) [inline overlord/snapstate/snapstate.go]: Actually, sorry, I take that back.. no need to check. It's active, so there must be something in Sequence or there's something seriously broken.

File: overlord/snapstate/snapstate.go [hunk 2]
```diff
@@ -285,3 +285,27 @@ func Set(s *state.State, name string, snapst *SnapState) {
 	snaps[name] = &raw
 	s.Set("snaps", snaps)
 }
+
+// All returns information about all active snaps.
+func All(s *state.State) ([]*snap.Info, error) {
```
Comment (niemeyer) [inline overlord/snapstate/snapstate.go]: `All` => `ActiveInfos`

This is slightly different, but it's okay. We can have it for now anyway.

Comment (zyga) [inline overlord/snapstate/snapstate.go]: Sure. As we settle all the other changes we can look at how things are being built and perhaps just make All and base ActiveInfos on top.

File: overlord/snapstate/snapstate.go (hunk 1 above)
Comment (zyga) [inline overlord/snapstate/snapstate.go]: Sure, it's one of those places that a panic is appropriate :-)

**Pull request merged**